Core/src/IfcGeometryConverter/FaceConverter.h

```diff
@@ -1486,6 +1486,60 @@ namespace OpenInfraPlatform {
 					}
 				}
 
+
+
+				void convertIfcPlanarExtent(const EXPRESSReference<typename IfcEntityTypesT::IfcPlanarExtent> planarExtent,
+					const carve::math::Matrix& pos,
+					std::shared_ptr<ItemData>& itemData)
+					const noexcept(false)
+				{
+					std::vector<carve::geom::vector<3>> loops;
+					std::vector<carve::geom::vector<3>> segment_start_points;
+					convertIfcPlanarExtent(planarExtent, loops, segment_start_points);
+
+					std::shared_ptr<carve::input::PolylineSetData> polylineData = std::make_shared<carve::input::PolylineSetData>();
+					polylineData->beginPolyline();
+					for (int i = 0; i < loops.size(); ++i) {
+						polylineData->addVertex(pos * loops.at(i));
+						polylineData->addPolylineIndex(i);
+					}
+					itemData->polylines.push_back(polylineData);
+				}
+
+
+				void convertIfcPlanarExtent(const EXPRESSReference<typename IfcEntityTypesT::IfcPlanarExtent> planarExtent,
+					std::vector<carve::geom::vector<3>>& loops,
+					std::vector<carve::geom::vector<3>>& segmentStartPoints)
+					const noexcept(false) 
+				{
+					double 	sizeInX = planarExtent->SizeInX;
+					double 	sizeInY = planarExtent->SizeInY;
+
+					//An extent is a rectangle specified by providing the coordinate of the lower left corner and the coordinate of the upper right corner in map units.
+					carve::geom::vector<3> lowerLeftCorner = carve::geom::VECTOR(-sizeInX, -sizeInY, 0.);
+					carve::geom::vector<3> upperRightCorner = carve::geom::VECTOR(sizeInX, sizeInY, 0.);
+
+					//I am not sure, if ItemData will take only 2 points of the rectangle, so I will add other 2 points here. 
+					carve::geom::vector<3> lowerRightCorner = carve::geom::VECTOR(sizeInX, -sizeInY, 0.);
+					carve::geom::vector<3> upperLeftCorner = carve::geom::VECTOR(-sizeInX, sizeInY, 0.);
+
+					//IfcPlanarBox
+					EXPRESSReference<typename IfcEntityTypesT::IfcPlanarBox> planarBox =
+						planarExtent.as<typename IfcEntityTypesT::IfcPlanarBox>();
+
+					carve::math::Matrix conicPositionMatrix = placementConverter->convertIfcAxis2Placement(planarBox->Placement);
+					carve::math::Matrix inverseConicPositionMatrix = GeomUtils::computeInverse(conicPositionMatrix);
+
+
+					loops.push_back(inverseConicPositionMatrix * lowerLeftCorner);
+					loops.push_back(inverseConicPositionMatrix * lowerRightCorner);
+					loops.push_back(inverseConicPositionMatrix * upperRightCorner);
+					loops.push_back(inverseConicPositionMatrix * upperLeftCorner);
+
+					segmentStartPoints.push_back(inverseConicPositionMatrix * lowerLeftCorner);
+				}

```

Comment:
I am not sure, If I have done it correctly, maybe I should do it like I did in IndexedPolygonalFace (create a triangulated list)

Reply:
Resolved: b4bb324